pkg/crinit/init/init_test.go

```diff
@@ -18,1576 +18,491 @@ package init
-			entKeyPairs.controllerManager.Cert.Raw,
-		},
-		PrivateKey: entKeyPairs.controllerManager.Key,
-	}
-
-	sCfg := &tls.Config{
-		Certificates:       []tls.Certificate{serverCert},
-		RootCAs:            roots,
-		ClientAuth:         tls.RequireAndVerifyClientCert,
-		ClientCAs:          roots,
-		InsecureSkipVerify: false,
-	}
-
-	cCfg := &tls.Config{
-		Certificates: []tls.Certificate{cmCert},
-		RootCAs:      roots,
-	}
-
-	return sCfg, cCfg, nil
-}
-
-func copyTLSConfig(cfg *tls.Config) *tls.Config {
-	// We are copying only the required fields.
-	return &tls.Config{
-		Certificates:       cfg.Certificates,
-		RootCAs:            cfg.RootCAs,
-		ClientAuth:         cfg.ClientAuth,
-		ClientCAs:          cfg.ClientCAs,
-		InsecureSkipVerify: cfg.InsecureSkipVerify,
-	}
-}
-
-func getEndpoint(apiserverServiceType v1.ServiceType, lbIP, advertiseAddress string) string {
-	endpoint := lbIP
-	if apiserverServiceType == v1.ServiceTypeNodePort {
-		if advertiseAddress != "" {
-			endpoint = advertiseAddress + ":" + strconv.Itoa(nodePort)
-		} else {
-			endpoint = nodeIP + ":" + strconv.Itoa(nodePort)
-		}
-	}
-	return endpoint
-}
-
-// TODO: Reuse the function addDNSProviderConfig once that function is converted to use versioned objects.
-func addDNSProviderConfigTest(dep *v1beta1.Deployment, secretName string) *v1beta1.Deployment {
-	const (
-		dnsProviderConfigVolume    = "config-volume"
-		dnsProviderConfigMountPath = "/etc/federation/dns-provider"
-	)
-
-	// Create a volume from dns-provider secret
-	volume := v1.Volume{
-		Name: dnsProviderConfigVolume,
-		VolumeSource: v1.VolumeSource{
-			Secret: &v1.SecretVolumeSource{
-				SecretName: secretName,
-			},
-		},
-	}
-	dep.Spec.Template.Spec.Volumes = append(dep.Spec.Template.Spec.Volumes, volume)
-
-	// Mount dns-provider secret volume to controller-manager container
-	volumeMount := v1.VolumeMount{
-		Name:      dnsProviderConfigVolume,
-		MountPath: dnsProviderConfigMountPath,
-		ReadOnly:  true,
-	}
-	dep.Spec.Template.Spec.Containers[0].VolumeMounts = append(dep.Spec.Template.Spec.Containers[0].VolumeMounts, volumeMount)
-	dep.Spec.Template.Spec.Containers[0].Command = append(dep.Spec.Template.Spec.Containers[0].Command, fmt.Sprintf("--dns-provider-config=%s/%s", dnsProviderConfigMountPath, secretName))
-
-	return dep
-}
-
-// TODO: Reuse the function addCoreDNSServerAnnotation once that function is converted to use versioned objects.
-func addCoreDNSServerAnnotationTest(deployment *v1beta1.Deployment, dnsZoneName, dnsProviderConfig string) (*v1beta1.Deployment, error) {
-	var cfg coredns.Config
-	if err := gcfg.ReadFileInto(&cfg, dnsProviderConfig); err != nil {
-		return nil, err
-	}
-
-	deployment.Annotations[util.FedDNSZoneName] = dnsZoneName
-	deployment.Annotations[util.FedNameServer] = cfg.Global.CoreDNSEndpoints
-	deployment.Annotations[util.FedDNSProvider] = util.FedDNSProviderCoreDNS
-	return deployment, nil
-}
+//func TestGenerateCredentials(t *testing.T) {
```

Comment:
Remove unused/incomplete code from here to line 508. You can always bring it back later?

Reply:
Agreed. Removed.